frontend/packages/ceph-storage-plugin/src/components/block-pool/create-block-pool.tsx

```diff
@@ -0,0 +1,171 @@
+  blockPoolReducer,
+  blockPoolInitialState,
+  BlockPoolActionType,
+  getPoolKindObj,
+  getErrorMessage,
+} from '../../utils/block-pool';
+
+import './create-block-pool.scss';
+
+const CreateBlockPool: React.FC<CreateBlockPoolProps> = ({ match }) => {
+  const {
+    params: { ns, appName },
+    url,
+  } = match;
+  const { t } = useTranslation();
+
+  const csvResource = {
+    kind: referenceForModel(ClusterServiceVersionModel),
+    name: appName,
+    namespace: ns,
+    isList: false,
+  };
+
+  const [state, dispatch] = React.useReducer(blockPoolReducer, blockPoolInitialState);
+  const [csv] = useK8sWatchResource<ClusterServiceVersionKind>(csvResource);
+  const [cephClusters, isLoaded, loadError] = useK8sWatchResource<CephClusterKind[]>(
+    cephClusterResource,
+  );
+
+  const memoizedCSV: K8sResourceKind = useDeepCompareMemoize(csv, true);
+  const cephCluster: CephClusterKind = useDeepCompareMemoize(cephClusters[0], true);
+
+  const storageClusterListPage = `${resourcePathFromModel(
+    ClusterServiceVersionModel,
+    appName,
+    ns,
+  )}/${referenceForModel(CephBlockPoolModel)}`;
+
+  const onClose = () => {
+    history.push(storageClusterListPage);
+  };
+
+  // Create new pool
+  React.useEffect(() => {
+    if (state.isSubmitted) {
+      dispatch({ type: BlockPoolActionType.SET_IS_SUBMITTED, payload: false });
+      const poolObj: StoragePoolKind = getPoolKindObj(state);
+
+      dispatch({ type: BlockPoolActionType.SET_INPROGRESS, payload: true });
+      k8sCreate(CephBlockPoolModel, poolObj)
+        .then(() => history.push(`${storageClusterListPage}/${state.poolName}`))
+        .finally(() => dispatch({ type: BlockPoolActionType.SET_INPROGRESS, payload: false }))
+        .catch((err) =>
+          dispatch({
+            type: BlockPoolActionType.SET_ERROR_MESSAGE,
+            payload: getErrorMessage(err.message) || 'Could not create block pool.',
+          }),
+        );
+    }
+  }, [state, state.isSubmitted, storageClusterListPage]);
+
+  if (cephCluster?.metadata.name === CEPH_EXTERNAL_CR_NAME) {
+    return (
+      <Modal
+        title={t('ceph-storage-plugin~Create Block Pool')}
+        titleIconVariant="warning"
+        isOpen
+        onClose={onClose}
+        variant="small"
+        isFullScreen={false}
+        actions={[
+          <Button key="confirm" variant="primary" onClick={onClose}>
+            {t('ceph-storage-plugin~Close')}
+          </Button>,
+        ]}
+      >
+        <strong>
+          {t(
+            'ceph-storage-plugin~Pool creation is not available for openshift container storage external mode.',
+          )}
+        </strong>
+      </Modal>
+    );
+  }
+
+  return (
+    <>
+      <div className="co-create-operand__header">
+        <div className="co-create-operand__header-buttons">
+          {memoizedCSV !== null && (
+            <BreadCrumbs
+              breadcrumbs={[
+                {
+                  name: memoizedCSV.spec?.displayName,
```

Comment:
No issue in hardcoding the name . because when the csv is not loaded then it moes right to top and does not looks good.

```suggestion
                  name: 'Openshift Container Storage',
```